### PR TITLE
Initial seed

### PR DESCRIPTION
**Pull request by jlprat**
Issue: akka/akka-http#1137
Use Variables for:
- Project name
- Project description
- Akka HTTP version
- Akka version

Contains code for:
- Normal boot, with routes defined in different files
- HttpApp
- Tests
- Information about template in README

### CODE REVIEW

File: src/main/g8/build.sbt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      "com.typesafe.akka" %% "akka-http"         % akkaHttpVersion,
+      "com.typesafe.akka" %% "akka-stream"       % akkaVersion,
+      "com.typesafe.akka" %% "akka-http-testkit" % akkaHttpVersion % Test,
+      "org.scalatest"     %% "scalatest"         % "3.0.1"         % Test,
+      "junit"              % "junit"             % "4.12"          % Test,
+      "com.novocode"       % "junit-interface"   % "0.11"          % Test
+    ),
+
+    testOptions += Tests.Argument(TestFrameworks.JUnit, "-v")
+  )
```
Comment (patriknw) [inline src/main/g8/build.sbt]: shall we ship the java one with maven and gradle build files also?
you can copy from https://github.com/akka/akka-java-seed.g8/pull/2/files#diff-2b2b73e47e4083fd40c15ae3964baf1f

Comment (jlprat) [reply src/main/g8/build.sbt]: :+1:

File: README.md [hunk 2]
```diff
@@ -1 +1,62 @@
-# akka-http-java-seed.g8
\ No newline at end of file
+A [Giter8][g8] template for Akka HTTP in Java
+
+Prerequisites:
+- JDK 8
+- sbt 0.13.13 or higher
```
Comment (ktoso) [inline README.md]: add a link to http://www.scala-sbt.org/download.html - java people may not know about it

File: README.md [hunk 3]
```diff
@@ -1 +1,62 @@
...
+the application:
+```
+mvn exec:exec
+```
+or
+```
+gradle run
+```
+Both commands run `com.example.HttpServer` by default.
+
```
Comment (ktoso) [inline README.md:52]: excellent to include gradle/maven instructions and files, thanks!

File: src/main/g8/build.gradle [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+repositories {
+    mavenCentral()
+    mavenLocal()
+}
+
+dependencies {
+    compile 'com.typesafe.akka:akka-http_2.12:$akka_http_version$'
+    compile 'com.typesafe.akka:akka-stream_2.12:$akka_version$'
+    testCompile 'com.typesafe.akka:akka-http-testkit_2.12:$akka_http_version$'
+    testCompile 'org.scalatest:scalatest_2.12:3.0.1'
```
Comment (ktoso) [inline src/main/g8/build.gradle:14]: is this dependency needed required?

**Review by ktoso — APPROVED** [review]
Looking good, some small question in line :)

File: README.md (hunk 2 above)
Comment (jlprat) [reply README.md]: Rigth!

File: src/main/g8/build.gradle (hunk 4 above)
Comment (jlprat) [reply src/main/g8/build.gradle:14]: I think I tried without and it blows up without it. I will recheck.

Comment (jlprat) [reply src/main/g8/build.gradle:14]: Without scalatest I get this error:

```
[info] [info] Compiling 3 Java sources to /tmp/sbt_823bc44b/scripted/target/scala-2.12/test-classes...
[info] [error] /tmp/sbt_823bc44b/scripted/src/test/java/com/example/HttpServerHttpAppTest.java:10: cannot access org.scalatest.junit.JUnitSuiteLike
[info] [error]   class file for org.scalatest.junit.JUnitSuiteLike not found
[info] [error] public class HttpServerHttpAppTest extends JUnitRouteTest {
[info] [error] /tmp/sbt_823bc44b/scripted/src/test/java/com/example/HttpServerHttpAppTest.java:17: cannot find symbol
[info] [error]   symbol:   method testRoute(akka.http.javadsl.server.Route)
[info] [error]   location: class com.example.HttpServerHttpAppTest
[info] [error]     appRoute = testRoute(server.route());
[info] [error] (test:compileIncremental) javac returned nonzero exit code
```

Maybe it is worth creating a ticket on akka/akka-http about this, WDYT?

Comment (ktoso) [reply src/main/g8/build.gradle:14]: Hmm, yeah I wonder - would be good to not have to not have java developers have to include scalatest explicitly; Ticket please I think

Comment (jlprat) [reply src/main/g8/build.gradle:14]: Done https://github.com/akka/akka-http/issues/1147

**Review by ktoso — APPROVED** [review]
This is looking good - thanks a ton :)
I propose to merge and we can always improve from thereon

**Pull request merged**